### PR TITLE
CI updates

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

**Comment by str4d** [comment]
The CI failures are because we need an MSRV bump to build with the `alloc` feature, due to `blake2b_simd 1.0.1` raising its MSRV to 1.59. We will do this bump separately from this PR, for `pasta_curves 0.6.0`.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,210 +4,87 @@ on: [push, pull_request]
...
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: --all-features --all-targets -- -W clippy::all
-
   codecov:
     name: Code coverage
     runs-on: ubuntu-latest
-
+    container:
+      image: xd009642/tarpaulin:develop-nightly
+      options: --security-opt seccomp=unconfined
```
Comment (daira) [inline .github/workflows/ci.yml:66]: Why is `seccomp=unconfined` needed?

**Review by daira — APPROVED** [review]
utACK, modulo the question about the seccomp option.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (daira) [reply .github/workflows/ci.yml:66]: Eh, it's fine I guess.

**Review by daira — APPROVED** [review]
utACK

**Pull request merged**